File: agents/api/metrics.py
from datetime import datetime


START_TIME = datetime.utcnow()

metrics = {
    "requests": 0,
    "providers": {}
}


def record_request(provider):

    metrics["requests"] += 1

    if provider not in metrics["providers"]:
        metrics["providers"][provider] = 0

    metrics["providers"][provider] += 1



def get_metrics():

    uptime = (
        datetime.utcnow() - START_TIME
    ).total_seconds()


    return {
        "requests": metrics["requests"],
        "providers": metrics["providers"],
        "uptime_seconds": uptime
    }

File: apps/web/src/api/metrics.ts
import { authHeaders } from "./session"

const API_URL =
  import.meta.env.VITE_ROUTER_API_URL ||
  "http://127.0.0.1:8000"


export type Metrics = {
  requests: number
  uptime_seconds: number
  providers: Record<string, number>
}


export async function getMetrics(): Promise<Metrics> {

  const response =
    await fetch(
      `${API_URL}/metrics`,
      {
        headers: {
          ...authHeaders()
        }
      }
    )

  return response.json()
}
